### PR TITLE
CloudServices Plugin should allow for registering multiple token endpoints

### PR DESCRIPTION
**Pull request by niegowski**
### Suggested merge commit message ([convention](https://ckeditor.com/docs/ckeditor5/latest/framework/guides/contributing/git-commit-message-convention.html))

Internal (cloud-services): CloudServices Plugin should allow for registering multiple token endpoints. 

---

### Additional information

### CODE REVIEW

File: packages/ckeditor5-cloud-services/src/cloudservices.js [hunk 1]
```diff
@@ -38,6 +39,14 @@ export default class CloudServices extends ContextPlugin {
 			this[ optionName ] = options[ optionName ];
 		}
 
+		/**
+		 * Map of `Token` object instances keyed by `tokenUrl`s.
+		 *
+		 * @private
+		 * @type {Map.<String, module:cloud-services-core/token~Token>}
```
Comment (ma2ciek) [inline packages/ckeditor5-cloud-services/src/cloudservices.js:46]: This accepts a string or a function as the key (which is actually a bit odd :smile:)

File: packages/ckeditor5-cloud-services/src/cloudservices.js [hunk 2]
```diff
@@ -67,9 +76,61 @@ export default class CloudServices extends ContextPlugin {
...
-		this.token = new CloudServices.Token( this.tokenUrl );
+		return this.registerTokenUrl( this.tokenUrl, true );
+	}
+
+	/**
+	 * Registers an additional authentication token URL for CKEditor Cloud Services or a callback to the token value promise. See the
+	 * {@link module:cloud-services/cloudservices~CloudServicesConfig#tokenUrl} for more details.
+	 *
+	 * @param {String|Function} tokenUrl The authentication token URL for CKEditor Cloud Services or a callback to the token value promise.
+	 * @param {Boolean} [isDefault=false] Whether this should be a default authentication token provider.
```
Comment (ma2ciek) [inline packages/ckeditor5-cloud-services/src/cloudservices.js:87]: It actually doesn't make sense to expose this flag as only the default value from the plugin initialization can use it.

So I'd try to remove the additional handling of the default token from this function.

File: packages/ckeditor5-cloud-services/src/cloudservices.js (hunk 1 above)
Comment (niegowski) [reply packages/ckeditor5-cloud-services/src/cloudservices.js:46]: Unfortunately, then it displays in docs like this: 

```
Map.&lt(String|Function), module:cloud-services-core/token~Token>
```

Comment (niegowski) [reply packages/ckeditor5-cloud-services/src/cloudservices.js:46]: but with current notation it's:

Comment (niegowski) [reply packages/ckeditor5-cloud-services/src/cloudservices.js:46]:
```
Map.<String, Token>
```

and `Token` is linking

File: packages/ckeditor5-cloud-services/src/cloudservices.js [hunk 3]
```diff
@@ -67,9 +76,61 @@ export default class CloudServices extends ContextPlugin {
...
+	 * @returns {Promise.<module:cloud-services-core/token~Token>}
+	 */
+	registerTokenUrl( tokenUrl, isDefault = false ) {
+		if ( this._tokens.has( tokenUrl ) ) {
+			return Promise.resolve( this.getTokenFor( tokenUrl ) );
```
Comment (ma2ciek) [inline packages/ckeditor5-cloud-services/src/cloudservices.js:92]: Should not it throw an error here?

**Review by ma2ciek — CHANGES_REQUESTED** [review]
A few minor comments.

File: packages/ckeditor5-cloud-services/src/cloudservices.js (hunk 3 above)
Comment (niegowski) [reply packages/ckeditor5-cloud-services/src/cloudservices.js:92]: Nope, it's reusing `Token` instance registered by the other feature - same `tokenUrl` = same token.

File: packages/ckeditor5-cloud-services/src/cloudservices.js (hunk 1 above)
Comment (ma2ciek) [reply packages/ckeditor5-cloud-services/src/cloudservices.js:46]: Ok. Let's go with this option then.

File: packages/ckeditor5-cloud-services/src/cloudservices.js (hunk 3 above)
Comment (niegowski) [reply packages/ckeditor5-cloud-services/src/cloudservices.js:92]: I'll add code comment there

Comment (ma2ciek) [reply packages/ckeditor5-cloud-services/src/cloudservices.js:92]: Ok. This makes sense if two different features share the same configuration.

**Review by ma2ciek — APPROVED** [review]

**Pull request merged**